### ISSUE TITLE
Auto Leveling (G29) with Marlin

### ISSUE THREAD

**Issue by adsaxton** [markniu/Bed_Distance_sensor#48]
I designed my own mount for the BDSensor, locating it 35.5mm behind and about 1.2mm left of the nozzle. My printers been adjusted to home Y to -30. The BDS calibrated and homes fine, but I haven't figured out how to tram the bed yet. When I try to auto-level with G29, it scans the 1st and 2nd lines and then freezes Marlin. I have to cycle the power to get the printer back. I'm at a loss since no error is reported.

**Comment by markniu** [comment]
did you try to comment out the BD_SENSOR_PROBE_NO_STOP ?  or please send me your configure files to me niujl123@sina.com

**Comment by adsaxton** [comment]
No. I did not (yet). I'll try that today. Here are my config files: 
[Configs.zip](https://github.com/markniu/Bed_Distance_sensor/files/13490089/Configs.zip)

I've also got the issue where, if I try to print using the "Real Time Leveling" feature, my printer insists on starting the first layer at 0.0mm (against the bed) rather that the intended 0.2mm. I have not figured out how to overcome this issue.

**Comment by adsaxton** [comment]
I found the problem! It appears there is a bug in the current version of Marlin-bugfix-2.1.x (02010300) where it uses Z_STOP_PIN, rather than Z_MIN_PROBE_PIN in some parts of the leveling code, including "Real Time Leveling". Changing Z_STOP_PIN to the same as Z_MIN_PROBE_PIN resolved the issues.

**Comment by Ahuffins** [comment]
> I found the problem! It appears there is a bug in the current version of Marlin-bugfix-2.1.x (02010300) where it uses Z_STOP_PIN, rather than Z_MIN_PROBE_PIN in some parts of the leveling code, including "Real Time Leveling". Changing Z_STOP_PIN to the same as Z_MIN_PROBE_PIN resolved the issues.

I am experiencing the same problem and was wondering where exactly do I make this change

**Comment by adsaxton** [comment]
> I am experiencing the same problem and was wondering where exactly do I make this change

It's in the PIN file. For my BTT SKR Mini E3:

`#define Z_STOP_PIN                          PC14  // PC2   // Z-STOP`

**Comment by ibrahimhakkicandan** [comment]
im experiencing the same issue. it goes all the way to the edge and freezes. my printer ender 3 s1 pro and in the pin files there is nothing related even 'Z'.
